apps/cowswap-frontend/src/modules/tradeFormValidation/hooks/useTokenCustomTradeError.ts

```diff
@@ -0,0 +1,36 @@
+import { useMemo } from 'react'
+
+import { useIsAnyOfTokensOndo } from '@cowprotocol/tokens'
+import { Currency } from '@uniswap/sdk-core'
+
+import { isValidQuoteError, QuoteApiErrorCodes } from 'api/cowProtocol/errors/QuoteError'
+
+import { TradeQuoteState } from '../../tradeQuote'
+
+export function useTokenCustomTradeError(
+  inputCurrency: Currency | undefined | null,
+  outputCurrent: Currency | undefined | null,
+  error: TradeQuoteState['error'],
+): string | undefined {
+  const inputToken = inputCurrency?.isToken ? inputCurrency : undefined
```

Comment:
Just to a bit optimize and do not run `useIsAnyOfTokensOndo` in the most of the cases:

```
  const isWeekendDay = isWeekend()
  const inputToken = !isWeekendDay && inputCurrency?.isToken ? inputCurrency : undefined
  const outputToken =  !isWeekendDay && outputCurrent?.isToken ? outputCurrent : undefined
  const ondoToken = useIsAnyOfTokensOndo(inputToken, outputToken)

```